src/aosm/azext_aosm/common/registry.py

```diff
@@ -0,0 +1,497 @@
+            uploadacr.azurecr.io/samples/nginx:stable
+        """
+
+        self._login()
+        try:
+            super().pull_image_to_local_registry(source_image)
+        except BadRequestError as error:
+            raise error
+        finally:
+            logger.info("Logging out of source registry %s", self.registry_name)
+
+            docker_logout_cmd = [
+                str(shutil.which("docker")),
+                "logout",
+                self.registry_name,
+            ]
+            call_subprocess_raise_output(docker_logout_cmd)
+
+    def copy_image_to_target_acr(
+        self,
+        source_image: str,
+        image_name: str,
+        image_version: str,
+        target_acr: str,
+    ) -> None:
+        """
+        Copy an image from this registry to a target ACR.
+
+        :param source_image: The image to copy
+            e.g. uploadacr.azurecr.io/samples/nginx:1.0.0
+        :param image_name: The name of the image
+            e.g. nginx
+        :param image_version: The version of the artifact
+            e.g. 1.0.0
+        :param target_acr: The target ACR
+        """
+
+        try:
+            print("Copying artifact from source registry")
+            # In order to use az acr import cross subscription, we need to use a token
+            # to authenticate to the source registry. This is documented as the way to
+            # us az acr import cross-tenant, not cross-sub, but it also works
+            # cross-subscription, and meant we didn't have to make a breaking change to
+            # the format of input.json. Our usage here won't work cross-tenant since
+            # we're attempting to get the token (source) with the same context as that
+            # in which we are creating the ACR (i.e. the target tenant)
+            get_token_cmd = [str(shutil.which("az")), "account", "get-access-token"]
+            # Dont use call_subprocess_raise_output here as we don't want to log the
+            # output
+            called_process = subprocess.run(  # noqa: S603
+                get_token_cmd,
+                encoding="utf-8",
+                capture_output=True,
+                text=True,
+                check=True,
+            )
+            access_token_json = json.loads(called_process.stdout)
+            access_token = access_token_json["accessToken"]
+        except subprocess.CalledProcessError as get_token_err:
+            # This error is thrown from the az account get-access-token command
+            # If it errored we can log the output as it doesn't contain the token
+            logger.debug(get_token_err, exc_info=True)
+            raise ClientRequestError(  # pylint: disable=raise-missing-from
+                "Failed to import image: could not get an access token from your"
+                " Azure account. Try logging in again with `az login` and then re-run"
+                " the command. If it fails again, please raise an issue and try"
```

Comment:
I suspect we shouldn't be using "raise an issue" type language. We should find out what we should use, but not necessarily as part of this story. Could you add a TODO above the `raise` line, though, please?